src/Moryx/StateMachines/StateBase.cs

```diff
@@ -99,11 +99,21 @@ internal static void Create<TStateBase>(IStateContext context, int? initialKey)
             where TStateBase : StateBase
```

Comment:
Correct me if I missunderstand something here, but this line makes already sure, that `StateBase` is assignable from `var stateBaseType = typeof(TStateBase);`, doesn't it? 🤔  
And I see no other changes in your PR aside from that, or am I overlooking something?